cedar-policy/src/api.rs

```diff
@@ -1237,6 +1314,44 @@ pub enum SchemaError {
     /// Support for this escape form has been dropped.
     #[error("schema contained the non-supported `__expr` escape")]
     ExprEscapeUsed,
+    /// Errors parsing the natural schema syntax.
+    #[error(transparent)]
+    #[diagnostic(transparent)]
+    NaturalSyntaxError(#[from] NaturalSyntaxError),
+    /// IO errors when calling `from_reader`
+    #[error(transparent)]
+    IOError(#[from] std::io::Error),
+    /// Error converting a schema into natural syntax
+    #[error(transparent)]
+    #[diagnostic(transparent)]
+    ToNaturalSyntax(#[from] ToNaturalSyntaxError),
+}
```

Comment:
`SchemaError` isn't marked `non_exhaustive`, so adding variants is technically a breaking change :(